xbmc/interfaces/legacy/Addon.h

```diff
@@ -100,10 +100,157 @@ namespace XBMCAddon
 #ifdef DOXYGEN_SHOULD_USE_THIS
       ///
       /// \ingroup python_xbmcaddon
-      /// @brief \python_func{ xbmcaddon.Addon([id]).getSettings() }
+      /// @brief \python_func{ xbmcaddon.Addon([id]).supportsInstanceSettings() }
+      /// Returns true if the requested add-on supports several instances.
+      ///
+      /// @return Returns true if supported, false otherwise
+      ///
+      ///
+      ///-----------------------------------------------------------------------
+      /// @python_v21 New function added.
+      ///
+      ///
+      /// **Example:**
+      /// ~~~~~~~~~~~~~{.py}
+      /// ..
+      /// supportsInstances = self.Addon('pvr.iptvsimple').supportsInstanceSettings()
+      /// ..
+      /// ~~~~~~~~~~~~~
+      ///
+      supportsInstanceSettings();
+#else
+      bool supportsInstanceSettings();
+#endif
+
+#ifdef DOXYGEN_SHOULD_USE_THIS
+      ///
+      /// \ingroup python_xbmcaddon
+      /// @brief \python_func{ xbmcaddon.Addon([id]).getKnownInstanceIds() }
+      /// Returns the list of known available instance ids.
+      ///
+      /// @return List of used add-on instances
+      ///
+      ///
+      ///-----------------------------------------------------------------------
+      /// @python_v21 New function added.
+      ///
+      ///
+      /// **Example:**
+      /// ~~~~~~~~~~~~~{.py}
+      /// ..
+      /// usedInstances = self.Addon('pvr.iptvsimple').getKnownInstanceIds()
+      /// ..
+      /// ~~~~~~~~~~~~~
+      ///
+      getKnownInstanceIds();
+#else
+      std::vector<unsigned int> getKnownInstanceIds();
+#endif
+
+#ifdef DOXYGEN_SHOULD_USE_THIS
+      ///
+      /// \ingroup python_xbmcaddon
+      /// @brief \python_func{ xbmcaddon.Addon([id]).getFreeNewInstanceId() }
+      /// Get free add-on instance id to set new independent instance and settings about.
+      ///
+      /// @note After setting the new instance settings a @ref setInstanceIdState() call is required.
+      ///
+      /// @return Next usable and free instance id
+      ///
+      ///
+      ///-----------------------------------------------------------------------
+      /// @python_v21 New function added.
+      ///
+      ///
+      /// **Example:**
+      /// ~~~~~~~~~~~~~{.py}
+      /// ..
+      /// addon = xbmcaddon.Addon('pvr.iptvsimple')
+      /// newinstanceid = addon.getFreeNewInstanceId()
+      /// ..
+      /// ~~~~~~~~~~~~~
+      ///
+      getFreeNewInstanceId();
+#else
```

Comment:
Same here. Why not simply `XX = addon.InstanceId()`?
Cleaner

Reply:
Thats as `addon.instance()` I see really confusing, as I expect then a handling class and not an identifier as return.
Maybe another, but has not found a good name for it.